lib/matplotlib/artist.py

```diff
@@ -75,8 +75,8 @@ def draw_wrapper(artist, renderer):
                 renderer.stop_filter(artist.get_agg_filter())
             if artist.get_rasterized():
                 renderer._raster_depth -= 1
-            if (renderer._rasterizing and artist.figure and
-                    artist.figure.suppressComposite):
+            if (renderer._rasterizing and (fig := artist.get_figure(root=True)) and
+                    fig.suppressComposite):
```

Comment:
In vector outputs if you have multiple images or things you are pre-rasterizing you may want to rasterize them all into one image in the final output (which is good because it saves space if they overlap, the pathological example is 100 4kx4k images with alpha that are all exactly on top of each other, by pre-merging them you get a 100x size win!).  On the other hand, if you have a bunch of images that only slightly overlap or you want to be able to re-extract the individual layers later you do not want to merge them.  As both are reasonable we have an (esoteric) flag to control it 🤣 .

Reply:
Thank you for the explanation.